src/js/base/__tests__/UploadBar.test.js

```diff
@@ -1,13 +1,44 @@
+import userEvent from "@testing-library/user-event";
 import { UploadBar } from "../UploadBar";
+import { screen, waitFor } from "@testing-library/react";
 
 describe("<UploadBar />", () => {
-    it("should render", () => {
-        const wrapper = shallow(<UploadBar />);
-        expect(wrapper).toMatchSnapshot();
+    let props;
+    beforeEach(() => {
+        props = { message: "", onDrop: jest.fn(), validator: () => null };
     });
 
-    it("should render optional message", () => {
-        const wrapper = shallow(<UploadBar message="Foobar" />);
-        expect(wrapper).toMatchSnapshot();
+    it("should render RTL", () => {
+        renderWithProviders(<UploadBar {...props} />);
+        expect(screen.getByText("Drag file here to upload")).toBeInTheDocument();
+        expect(screen.getByRole("button", { name: "Browse Files" })).toBeInTheDocument();
+    });
+
+    it("should display optional message", () => {
+        props.message = "test_message";
+        renderWithProviders(<UploadBar {...props} />);
+        expect(screen.getByText("test_message")).toBeInTheDocument();
+    });
+
+    it("should call onDrop with uploaded files", async () => {
+        renderWithProviders(<UploadBar {...props} />);
+        const file = new File(["test"], "test_file.fa.gz", { type: "application/gzip" });
+        await userEvent.upload(screen.getByTestId("upload-input"), file);
```

Comment:
Can you not use input role here instead of test-id?

Reply:
Unfortunately no because `input` elements of `type=file` don't have a role. I originally used the test-id because this input element is invisible to the user, has no role, but must be targeted for the test. That said, I ended up changing it to use an aria-label as I could see the possibility that assistive technology may need to directly target the input element as well.